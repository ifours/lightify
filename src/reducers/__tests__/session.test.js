import reducer, { initialState } from '../session';

describe('Session reducer', () => {
  it('should return initialState', () => {
    const state = reducer(undefined, {});

    expect(state).toMatchSnapshot();
    expect(state).toBe(initialState);
  });

  it('should return new state for AUTH_TOKEN_RECEIVE action', () => {
    const payload = {
      accessToken: 'foo',
      expiresIn: 'bar',
      date: 'baz',
    };
    const state = reducer(undefined, { type: 'AUTH_TOKEN_RECEIVE', payload });

    expect(state).toMatchSnapshot();
  });

  it('should return initial state for AUTH_TOKEN_EXPIRE action', () => {
    const state = {
      accessToken: 'foo',
      expiresIn: 'bar',
      date: 'baz',
    };
    const newState = reducer(state, { type: 'AUTH_TOKEN_EXPIRE' });
    
    expect(newState).toMatchSnapshot();
    expect(newState).toBe(initialState);
  });
});
