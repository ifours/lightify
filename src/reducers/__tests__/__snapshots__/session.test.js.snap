// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Session reducer should return initial state for AUTH_TOKEN_EXPIRE action 1`] = `
Object {
  "accessToken": undefined,
  "date": 0,
  "expiresIn": 0,
}
`;

exports[`Session reducer should return initialState 1`] = `
Object {
  "accessToken": undefined,
  "date": 0,
  "expiresIn": 0,
}
`;

exports[`Session reducer should return new state for AUTH_TOKEN_RECEIVE action 1`] = `
Object {
  "accessToken": "foo",
  "date": "baz",
  "expiresIn": "bar",
}
`;
